src/components/library-app-2/book-form.js

```diff
@@ -1,218 +1,256 @@
-//           <SelectDropdown
-//             data={data}
-//             defaultButtonText="Select a publisher"
-//             onSelect={selectedItem => {
-//               handleFieldChange('publisher', selectedItem);
-//             }}
-//             buttonTextAfterSelection={selectedItem => {
-//               return selectedItem;
-//             }}
-//             rowTextForSelection={item => {
-//               return item;
-//             }}
-//             buttonStyle={styles.dropdownBtnStyle}
-//             buttonTextStyle={styles.dropdownBtnTxtStyle}
-//             renderDropdownIcon={isOpened => {
-//               return (
-//                 <FontAwesome
-//                   name={isOpened ? 'chevron-up' : 'chevron-down'}
-//                   color={'#444'}
-//                   size={18}
-//                 />
-//               );
-//             }}
-//             dropdownIconPosition={'right'}
-//             rowStyle={styles.dropdownRowStyle}
-//             rowTextStyle={styles.dropdownRowTxtStyle}
-//           />
-//         </View>
-//       ),
-//     },
-//     {label: 'Price', keyboardType: 'numeric', value: price, key: 'price'},
-//     {label: 'Email', keyboardType: 'email-address', value: email, key: 'email'},
-//     {label: 'Website', keyboardType: 'url', value: website, key: 'website'},
-//     {
-//       key: 'displayBook',
-//       Comp: () => (
-//         <View style={styles.checkboxContainer}>
-//           <CheckBox
-//             value={displayBook}
-//             onValueChange={value => handleFieldChange('displayBook', value)}
-//             style={styles.checkbox}
-//           />
-//           <Text style={styles.label}>
-//             Do you want to display this Book in library?
-//           </Text>
-//         </View>
-//       ),
-//     },
-//   ];
+function BookForm() {
+  const [formValues, setFormValues] = useState(defaultFormValues);
+  const [isSubmitting, setIsSubmitting] = useState(false);
 
-//   const handleFieldChange = (field, value) => {
-//     setFormValues(prevFormValues => ({...prevFormValues, [field]: value}));
-//   };
+  const {bookName, authorName, price, email, website, displayBook} = formValues;
 
-//   const handleSubmit = () => {};
+  const formFields = [
+    {
+      label: 'Book Name',
+      keyboardType: 'default',
+      value: bookName,
+      key: 'bookName',
+    },
+    {
+      label: 'Author Name',
+      keyboardType: 'default',
+      value: authorName,
+      key: 'authorName',
+    },
+    {
+      comp: (
+        <View key="publisher" style={styles.formGroup}>
+          <Text style={styles.label}>Publishers</Text>
+          <SelectDropdown
+            data={dropdownData}
+            defaultButtonText="Select a publisher"
+            onSelect={selectedItem => {
+              handleFieldChange('publisher', selectedItem);
+            }}
+            buttonTextAfterSelection={selectedItem => selectedItem}
```

Comment:
Cant this be a single separate function ?

Reply:
done